RELEASE.md

```diff
@@ -0,0 +1,61 @@
+# Release
+
+This document outlines the process for creating a new release for VMClarity using the [Go MultiMod Releaser](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/multimod). All code block examples provided below correspond to an update to version `v0.7.0`, please update accordingly.
+
+## 1. Update the New Release Version
+
+* Create a new branch for the release version update.
+```sh
+git checkout -b release/v0.7.0
+```
+
+* Modify the `versions.yaml` file to update the version for VMClarity's module-set. Keep in mind that the same version is applied to all modules.
+```
+  vmclarity:
+-    version: v0.6.0
++    version: v0.7.0
+```
+
+* Commit the changes with a suitable message.
+```sh
+git add versions.yaml
+git commit -m "release: update module set to version v0.7.0"
+```
+
+* Run the version verification command to check for any issues.
+```sh
+make multimod-verify
+```
+
+## 2. Bump All Dependencies to the New Release Version
+
+* Run the following command to update all `go.mod` files to the new release version.
+```sh
+make multimod-prerelease
+```
+
+* Review the changes made in the last commit to ensure correctness.
+
+* Push the branch to the GitHub repository.
+```sh
+git push origin release/v0.7.0
+```
+
+* Create a pull request with the changes.
+
+## 3. Create and Push Tags
+
+* After the pull request is approved and merged, update your local main branch.
+```sh
+git checkout main
+git pull origin main
+```
+
+* Create and push the tags for the last commit to the repository.
```

Comment:
This will create a tag and push it automatically based on the version YAML file, right? Would be useful to include a bit more context here about the flow (is the branch `release/{version}` needed, or the tag, or the version file?)

Reply:
thanks for reviewing, @ramizpolic!
> This will create a tag and push it automatically based on the version YAML file, right?

Yes, indeed and the YAML file is updated in the main branch at this point.

I will try to make this clearer while keeping the instructions simple but all these steps are required to the release.

Reply:
nvm, im blind and glanced over the actual make step, this is okay!

Reply:
are you sure? I agree that the steps must be as clear as possible to the user so let me know if I can improve the guide somehow :D 

I also thought about adding a troubleshooting section referring to https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/multimod/v0.12.0/multimod

Reply:
I think the tag push might be a bit problematic since it's automated by the script and does not have confirmation layer. Having a potential preview of what will be executed or some details could potentially be useful. I am not sure what the actual requirements for these scripts are and what can go wrong here, I am just worried if rewrites could happen (or some other issues) and if the checklist covers those scenarios.